### PR TITLE
Refactor tensor_utils.cpp.

### PR DESCRIPTION
**Pull request by nilaykp**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/15691

### Problem description
tensor_utils.cpp has duplication of code.

### What's changed
Remove common code.

### Checklist
- [ ] Post commit CI passes [Link](https://github.com/tenstorrent/tt-metal/actions/runs/12271307178)
- [ ] Blackhole Post commit (if applicable)
- [ ] Nightly fast dispatch [Link](https://github.com/tenstorrent/tt-metal/actions/runs/12271313777/job/34240428341) same as [main](https://github.com/tenstorrent/tt-metal/actions/runs/12270594879)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by TT-BrianLiu — COMMENTED** [review]
Clean up looks good! Should we move conv related functions to a separate file? Having them in a generic `tensor_utils` file seems misleading, since generic OP writers will never use these functions. Maybe we should even hide it under some `conv::detail` or `conv::utils` namespace.

File: ttnn/cpp/ttnn/tensor/tensor_utils.cpp [hunk 1]
```diff
@@ -32,6 +32,53 @@ Tensor convert_tensor(const Tensor& input_tensor, compute_& compute) {
...
+            auto output_packed_data =
+                output_dtype == DataType::BFLOAT8_B
+                    ? pack_fp32_vec_as_bfp8_tiles(output_float_data, /*row_major_input=*/false, /*is_exp_a=*/false)
+                    : pack_fp32_vec_as_bfp4_tiles(output_float_data, /*row_major_input=*/false, /*is_exp_a=*/false);
+            auto output_uint32_buffer = owned_buffer::create<uint32_t>(std::move(output_packed_data));
+            return Tensor(
+                std::move(OwnedStorage{std::move(output_uint32_buffer)}), output_shape, output_dtype, Layout::TILE);
+        }
+    } else {
+        TT_ASSERT((output_dtype != DataType::BFLOAT8_B) || (output_dtype != DataType::BFLOAT4_B));
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/tensor/tensor_utils.cpp:77]: Can we change this to fatal please?

**Review by ayerofieiev-tt — APPROVED** [review]
As the next step, please move conv related utils out as Brian suggested.

**Review by TT-BrianLiu — APPROVED** [review]

File: ttnn/cpp/ttnn/tensor/tensor_utils.cpp (hunk 1 above)
Comment (nkpatel-tt) [reply ttnn/cpp/ttnn/tensor/tensor_utils.cpp]: Done.

**Comment by nkpatel-tt** [comment]
> As the next step, please move conv related utils out as Brian suggested.

Created issue, https://github.com/tenstorrent/tt-metal/issues/15892

**Pull request merged**